### PR TITLE
Fix #2917, #2969: Build bazel for util/extensions and fix typo in versions.bzl

### LINKED ISSUE #2917: Create a BUILD.bazel file for the extensions package in the utility module
This task is part of the build system migration from Gradle to Bazel. 

## Prerequesites

- Make sure you are able to build Oppia using Bazel
- Follow the [Oppia Bazel Setup instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions) to setup Bazel and learn the terminology.
- Read the [Gradle Bazel Migration Best Practies and FAQ](https://github.com/oppia/oppia-android/wiki/Gradle--Bazel-Migration-Best-Practices-and-FAQ) for more context and see if there's an answer there to any question you have.

## Task

1. Create a BUILD.bazel file in

`src/main/java/org/oppia/android/util/extensions/`

2. Define the kt_android_library library/libraries

Based on the imports in the Kotlin files, add the required third_party dependencies to make the libraries build. Check the Useful links section for a list of third_party dependencies.

You can start with the following BUILD.bazel stub

```
"""
Add a description of the package here
"""

load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")

kt_android_library(
    name = "bundle_extensions",
    srcs = [
        "BundleExtensions.kt",
    ],
)

kt_android_library(
    name = "live_data_ext",
    srcs = [
        "LiveDataExt.kt",
    ],
)
```

> Note: live_data_ext might not be used anywhere.

3. Add the directory to the MIGRATED_PROD_FILES of the module's root BUILD.bazel file

4. Add the libraries to the list of deps dependencies in the targets that require the library

You will have to set the apropriate visbility. See the list of visibilities in the link below.

5. Clean up any unused deps dependencies

6. Verify that the app builds

## Useful links

- [BUILD.bazel files](https://github.com/oppia/oppia-android/search?q=filename%3ABUILD.bazel&type=code)
- [Visibilities](https://github.com/oppia/oppia-android/blob/76dc13dba3ad77aecd2218a5372949d6551f12fe/BUILD.bazel)
- [third_party dependencies](https://github.com/oppia/oppia-android/blob/a70d9b016f0d77dd30c4939041f82ea02ee28d1d/third_party/versions.bzl)

### LINKED ISSUE #2969: Typo in `versions.bzl`
There's a typo in a comment in the versions.bzl at line number 8 - https://github.com/oppia/oppia-android/blob/75c25b8094a9a84197397cd40887a491f8118d0f/third_party/versions.bzl#L8

Solution - 
Replace "mutli" with "multi".

### PR DESCRIPTION
**Pull request by Victor-Titan**
## Explanation

Fixes #2917. Created a BUILD.bazel file in the util/extensions directory to migrate the dependencies.
Fixes #2969. 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (FareesHussain) [inline .idea/misc.xml]: Please revert this file.

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@Victor-Titan PTAL at the 4th point of the checklist in the PR description
 - The PR does not contain any unnecessary auto-generated code from Android Studio.

File: .idea/misc.xml (hunk 1 above)
Comment (Victor-Titan) [reply .idea/misc.xml]: I used the command line to commit and this file always automatically formats itself so slipped my attention. Will revert.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "//third_party:com_google_protobuf_protobuf-lite",
+    ],
+)
+
+kt_android_library(
+    name = "live_data_ext",
+    srcs = [
+        "LiveDataExt.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
```
Comment (fsharpasharp) [inline utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Remove this visibility for now, since it doesn't seem to be used.

**Review by fsharpasharp — CHANGES_REQUESTED** [review]
Looks good. Thanks for adding the new dependency.

**Review by FareesHussain — APPROVED** [review]
LGTM, Thanks.

Can you confirm https://github.com/oppia/oppia-android/issues/2917#issuecomment-803233308 if this is correct

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 2 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Ok

**Comment by Victor-Titan** [comment]
> LGTM, Thanks.
> 
> Can you confirm [#2917 (comment)](https://github.com/oppia/oppia-android/issues/2917#issuecomment-803233308) if this is correct

That dependency worked for me and the project built successfully.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 2 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: @fsharpasharp PTAL now

**Review by FareesHussain — APPROVED** [review]

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel [hunk 3]
```diff
@@ -20,8 +20,4 @@ kt_android_library(
     srcs = [
         "LiveDataExt.kt",
     ],
-    visibility = ["//:oppia_api_visibility"],
-    deps = [
```
Comment (fsharpasharp) [inline utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: You should keep the dependency, so that the library still builds.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 2 above)
Comment (fsharpasharp) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Removing visibility looks good, but I left another comment about removing the dependency.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 3 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: The app built on running bazel build //:oppia so I figured we didn't need the dependency.

Comment (fsharpasharp) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: The app will run since the library is not used. But maybe we should remove this file? @BenHenning

Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: @fsharpasharp also, I'm not sure I added the correct dependency for that library as the library wasn't used anywhere so build failures did not occur.

Comment (fsharpasharp) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: You can cd to the extensions directory and run

`bazel build :all`

to build all libraries in the package.

Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: @fsharpasharp the libraries built successfully for the androidx_lifecycle_lifecycle-extensions dependency. PTAL

Comment (fsharpasharp) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Thanks! Leaving this thread up so that Ben can confirm if we can delete the file.

**Comment by Victor-Titan** [comment]
@BenHenning PTAL

**Review by prayutsu — COMMENTED** [review]
@Victor-Titan Not directly related to this PR, but can you please fix #2969 here?

**Comment by Victor-Titan** [comment]
> @Victor-Titan Not directly related to this PR, but can you please fix #2969 here?

Will do, should I assign the issue to myself?

**Comment by Victor-Titan** [comment]
@prayutsu PTAL

**Review by prayutsu — APPROVED** [review]
LGTM, thanks @Victor-Titan.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
+
+kt_android_library(
+    name = "bundle_extensions",
+    srcs = [
+        "BundleExtensions.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        "//third_party:com_google_protobuf_protobuf-lite",
```
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel:14]: Why is this needed as a new dependency? It seems like BundleExtensions was building fine before--what changed?

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    ],
+)
+
+kt_android_library(
+    name = "live_data_ext",
```
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Let's call it 'Extensions' for consistency. Please also rename ``LiveDataExt`` while here for consistency.

**Review by BenHenning — COMMENTED** [review]
Thanks @Victor-Titan! Just had a couple of comments.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 4 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel:14]: @BenHenning the BundleExtensions imports the protobuf package, and the project did not build for me without adding this dependency. We're gonna remove the model library so we needed individual dependencies that just imported
`import com.google.protobuf.ByteString`
`import com.google.protobuf.InvalidProtocolBufferException`
`import com.google.protobuf.MessageLite`

And this new dependency does that. For reference, look at [this](https://github.com/oppia/oppia-android/issues/2917#issuecomment-803129908) comment.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 5 above)
Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Done

**Comment by Victor-Titan** [comment]
@BenHenning PTAL

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 4 above)
Comment (BenHenning) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel:14]: Ah, thanks for clarifying. That makes sense.

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 3 above)
Comment (BenHenning) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Hmm I'm having difficulty tracking the origins of this file, but, given that it isn't being used, removing it sounds good.

Comment (Victor-Titan) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: @BenHenning So should I just delete that kotlin file and remove it's library from the bazel file then?

**Review by fsharpasharp — APPROVED** [review]
Thanks for making the changes!

File: utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel (hunk 3 above)
Comment (BenHenning) [reply utility/src/main/java/org/oppia/android/util/extensions/BUILD.bazel]: Yup, SGTM.

**Review by BenHenning — APPROVED** [review]
Thanks @Victor-Titan! LGTM.
